fix checking for confirmation message

atmonshi opened the pull request.

atmonshi committed:
fix checking for confirmation message

Abdulmajeed-Jamaan commented:
I know this issue is from filament RichEditor component setting the value to empty tag `<p></p>` when saving event if the content is empty.

I would prefer moving this logic from the render hook to pre saving hook (saving event in the model, or filament component hydaration hooks, or mutate data hook before save in the filament resource page), so it would be more proper and maintainable even the database layer set correctly.

Abdulmajeed-Jamaan committed:
Revert "fix checking for confirmation message"

This reverts commit 543a141605f963682effd1d7f841919cf0ac429e.

Abdulmajeed-Jamaan committed:
Add state dehydration to RichEditor fields

RichEditor fields for 'details' and 'options.confirmation-message' now use dehydrateStateUsing to ensure only non-empty, non-HTML-stripped content is saved. This prevents storing empty or HTML-only values.

Abdulmajeed-Jamaan committed:
Fix styling

Abdulmajeed-Jamaan committed:
Fix form details rendering condition

Replaces the use of filled(strip_tags(...)) with !empty(...) to correctly check if form details should be rendered. This ensures details are displayed when present.

Abdulmajeed-Jamaan committed:
Merge branch 'fix-confirmation-message' of https://github.com/lara-zeus/bolt into fix-confirmation-message
